src/domains/chart/components/lib-charts/dygraph-chart.tsx

```diff
@@ -1132,7 +1135,7 @@ export const DygraphChart = ({
       )}
       {alarm && hasLegend && (
         // @ts-ignore
-        <AlarmBadge ref={alarmBadgeRef} status={alarm.status} label={alarm.value} />
+        <AlarmBadge isVisible={isBadgeVisible} ref={alarmBadgeRef} status={alarm.status} label={alarm.value} />
```

Comment:
If for the above comment, we don't need the container to be rendered when not visible, I prefer to handle the visibility here, rather passing a flag in the component.

Reply:
Unfortunately we need to render Container in AlarmBadge, so we can use the ref to its DOM Node (in dygraph callback). If i had that ref unavailable at time of callback, i would need to keep params (applied to `ref.current.style`) in the state and i'm worried that it would make the logic behind it less clear.

I limited the nr of times `AlarmBadge` will need to render by having it under `alarm` conditional - it will only render when the current chart will have "show alarm mode". @novykh how does it sound?

Btw. probably this feature will be changed in next iteration